components/core/src/clp/Grep.cpp

```diff
@@ -540,76 +557,346 @@ std::optional<Query> Grep::process_raw_query(
+                std::string current_string = processed_search_string.substr(j, i - j + 1);
+                std::vector<QueryLogtype> suffixes;
+                clp::SearchToken search_token;
+                if (current_string == "*") {
+                    suffixes.emplace_back('*', "*", false);
+                } else {
+                    // Add * if preceding and proceeding characters are *
+                    bool prev_star = j > 0 && processed_search_string[j - 1] == '*';
+                    bool next_star = i < processed_search_string.back() - 1
+                                     && processed_search_string[i + 1] == '*';
+                    if (prev_star) {
+                        current_string.insert(0, "*");
+                    }
+                    if (next_star) {
+                        current_string.push_back('*');
+                    }
+                    bool is_surrounded_by_delims = false;
+                    if ((j == 0 || current_string[0] == '*'
+                         || forward_lexer.is_delimiter(processed_search_string[j - 1]))
+                        && (i == processed_search_string.size() - 1 || current_string.back() == '*'
+                            || forward_lexer.is_delimiter(processed_search_string[i + 1])))
+                    {
+                        is_surrounded_by_delims = true;
+                    }
+                    bool contains_wildcard = false;
+                    set<uint32_t> schema_types;
+                    // All variables must be surrounded by delimiters
+                    if (is_surrounded_by_delims) {
+                        log_surgeon::ParserInputBuffer parser_input_buffer;
+                        std::string regex_search_string;
+                        bool contains_central_wildcard = false;
+                        uint32_t pos = 0;
+                        for (char const& c : current_string) {
+                            if (c == '*') {
+                                contains_wildcard = true;
+                                regex_search_string.push_back('.');
+                                if (pos > 0 && pos < current_string.size() - 1) {
+                                    contains_central_wildcard = true;
+                                }
+                            } else if (log_surgeon::SchemaParser::get_special_regex_characters()
+                                               .find(c)
+                                       != log_surgeon::SchemaParser::get_special_regex_characters()
+                                                  .end())
+                            {
+                                regex_search_string.push_back('\\');
+                            }
+                            regex_search_string.push_back(c);
+                            pos++;
+                        }
+                        log_surgeon::NonTerminal::m_next_children_start = 0;
+                        log_surgeon::Schema schema2;
```

Comment:
Can we give this a more descriptive name?

Reply:
Updated to substring_schema